### ISSUE TITLE
Covidtestzentren - Datenbank

### ISSUE THREAD

**Issue by errotu** [eknoes/covidbot#152]
Eine zentrale, deutschlandweite Datenbank scheint es bisher nicht zu geben. Hier eine Übersicht der Webangebote aller Bundesländer. Kaum ein Bundesland bietet einen Datenexport an. Wir könnten aber natürlich einfach bei jedem Detailbericht zu einem Ort auf die Plattform des jeweiligen Bundeslandes verlinken.

- [Baden-Württemberg](https://www.kvbawue.de/index.php?id=1102); kein Datenexport/Schnittstelle
- [Bayern](https://www.stmgp.bayern.de/coronavirus/bayerische-teststrategie/); kein Datenexport/Schnittstelle
- [Berlin](https://test-to-go.berlin/); kein Datenexport/Schnittstelle
- [Brandenburg](https://brandenburg-testet.de/bb-testet/de/); kein Datenexport/Schnittstelle
- [Bremen](https://www.gesundheit.bremen.de/gesundheit/corona/corona_ambulanz-32720); kein Datenexport/Schnittstelle
- [Hamburg](https://www.hamburg.de/corona-schnelltest/); kein Datenexport/Schnittstelle
- [Hessen](https://www.corona-test-hessen.de/?action=Daten), mit **Datenexport/Schnittstelle**
- [Mecklenburg-Vorpommern](https://www.regierung-mv.de/Landesregierung/wm/Aktuelles--Blickpunkte/Wichtige-Informationen-zu-Corona-Virus-Testzentren/); kein Datenexport/Schnittstelle
- [Niedersachsen](https://www.niedersachsen.de/Coronavirus/Testunghinweise-zur-testung-auf-corona-198156.html#wo); kein Datenexport/Schnittstelle
- [Nordrhein-Westfalen](https://www1.wdr.de/nachrichten/themen/coronavirus/corona-schnelltest-kostenlos-teststellen-100.html); kein Datenexport/Schnittstelle
- [Rheinland-Pfalz](https://covid-19-support.lsjv.rlp.de/hilfe/covid-19-test-dashboard/); mit **Datenexport/Schnittstelle**
- [Saarland](https://www.saarland.de/DE/portale/corona/impfungtest/testzentrum/testmoeglichkeiten/schnelltestskommunen.html); kein Datenexport/Schnittstelle
- [Sachsen](https://www.coronavirus.sachsen.de/coronatests-in-sachsen-9448.html); mit **Datenexport/Schnittstelle** (nur xlsx)
- [Sachsen-Anhalt](https://coronavirus.sachsen-anhalt.de/angebote-fuer-schnelltests/suche-nach-testzentren/?no_cache=1); kein Datenexport/Schnittstelle
- [Schleswig-Holstein](https://www.schleswig-holstein.de/DE/Schwerpunkte/Coronavirus/Allgemeines/TeststationenKarte/teststationen_node.html?lang=de); kein Datenexport/Schnittstelle
- [Thüringen](https://www.tmasgff.de/covid-19/tests); kein Datenexport/Schnittstelle

**Comment by eknoes** [comment]
Oh je, bspw. bei NRW ist das ja sogar nur eine Liste zu den Zentren der Kommunen. Habe gerade noch für Hamburg diese API gefunden: https://suche.transparenz.hamburg.de/dataset/corona-testzentren-hamburg

**Comment by eknoes** [comment]
Teilweise gibt es nicht öffentliche Schnittstellen, bspw. bei Meck-Pomm: https://www.regierung-mv.de/gaialight-apps/testzentren/_gaia_json.php

**Comment by eknoes** [comment]
Wir könnten überlegen, die öffentlichen Schnittstellen zu nutzen, die nicht-öffentlich auch zu benutzen bzw. da die Zuständigen Behörden um Nutzungserlaubnis bitten, beim Rest crowdsourcen und dann eine eigene API zur Verfügung stellen mit allen gesammelten Daten.

**Comment by errotu** [comment]
Was meinst du mit "nicht öffentlich"? Eher nicht dokumentiert, oder? Der Zugriff ist ja möglich.

**Comment by errotu** [comment]
Ja, finde ich gut. Wollen wir das hier mal sammeln und wenn wir neue Schnittstellen finden als Änderung eintragen? Ist im CSV-Format:

**Comment by errotu** [comment]
Bundesland,URL,API,API_Format,API_URL
Baden-Württemberg,https://www.kvbawue.de/index.php?id=1102,No,,
Bayern,https://www.stmgp.bayern.de/coronavirus/bayerische-teststrategie/,No,,
Berlin,https://test-to-go.berlin/,No,,
Brandenburg,https://brandenburg-testet.de/bb-testet/de/,No,,
Bremen,https://www.gesundheit.bremen.de/gesundheit/corona/corona_ambulanz-32720,No,,
Hamburg,https://www.hamburg.de/corona-schnelltest/,Yes,JSON,https://suche.transparenz.hamburg.de/api/3/action/package_show?id=corona-testzentren-hamburg
Hessen,https://www.corona-test-hessen.de/?action=Daten,Yes,XML,https://www.corona-test-hessen.de/data/corona-test-hessen.xml?v=22836BA9C360BE1022B689D6DCA16813
Mecklenburg-Vorpommern,https://www.regierung-mv.de/Landesregierung/wm/Aktuelles--Blickpunkte/Wichtige-Informationen-zu-Corona-Virus-Testzentren/,Yes,JSON,https://www.regierung-mv.de/gaialight-apps/testzentren/_gaia_json.php
Niedersachsen,https://www.niedersachsen.de/Coronavirus/Testunghinweise-zur-testung-auf-corona-198156.html#wo,No,,
Nordrhein-Westfalen,https://www1.wdr.de/nachrichten/themen/coronavirus/corona-schnelltest-kostenlos-teststellen-100.html,No,,
Rheinland-Pfalz,https://covid-19-support.lsjv.rlp.de/hilfe/covid-19-test-dashboard/,Yes,CSV,No stable URL
Saarland,https://www.saarland.de/DE/portale/corona/impfungtest/testzentrum/testmoeglichkeiten/schnelltestskommunen.html,No,,
Sachsen,https://www.coronavirus.sachsen.de/coronatests-in-sachsen-9448.html,Yes,XLSX,https://www.coronavirus.sachsen.de/download/Testmoeglichkeiten.xlsx
Sachsen-Anhalt,https://coronavirus.sachsen-anhalt.de/angebote-fuer-schnelltests/suche-nach-testzentren/?no_cache=1,No,,
Schleswig-Holstein,https://www.schleswig-holstein.de/DE/Schwerpunkte/Coronavirus/Allgemeines/TeststationenKarte/teststationen_node.html?lang=de,No,,
Thüringen,https://www.tmasgff.de/covid-19/tests,No,,

**Comment by budda85** [comment]
Ich weiß nicht ob das hilft oder gesehen wurde, aber für NRW gibt es unterhalb der Grafik einen Link "Daten herunterladen". Da gibt es eine CSV zurück welche allerdings auch nur Links zu den Städten bzw. Kreisen enthält.

**Comment by errotu** [comment]
Vielen Dank für den Hinweis! Ich habe es mal [hier](https://github.com/eknoes/covidbot/blob/main/resources/testzentren.csv) in die Tabelle eingetragen. Leider gibt es auch keine stabile URL auf die CSV-Datei, so dass das automatisierte Abrufen eher schwer werden dürfte.
